org.opent2t.sample.lamp.superpopular/com.wink.lightbulb/js/thingTranslator.js

```diff
@@ -44,107 +51,476 @@ function findResource(schema, di, resourceId) {
 }
 
 /**
- * Wink does not always populate every desired_state property, but last_reading doesn't necessarily
- * update as soon as we send our PUT request. Instead of relying just on one state or the other,
- * we use this StateReader class to read from desired_state if it is there, and fall back to last_reading
- * if it is not.
+ * Validates that an object (arguments[0]) contains properties
+ * (arguments[1...n]) and throws a descriptive OpenT2T error if the
+ * request is malformed.
+ * 
+ * TODO: This method should be moved to a shared location for all translators
```

Comment:
No, I simply wanted to tag which methods I'm going to be pulling out into a shared library at some point in the near future.

---
In reply to: [106017868](https://github.com/openT2T/translators/pull/209#discussion_r106017868) [](ancestors = 106017868)